### PR TITLE
Auto/map based prediction

### PR DESCRIPTION
**Pull request by Sugatyon**
## Description

autoware_lanelet2_msgs/msg/map_bin.hpp
→
autoware_auto_mapping_msgs/msg/had_map_bin.hpp

input
autoware_perception_msgs/msg/dynamic_object_array.hpp
→
autoware_auto_perception_msgs/msg/tracked_objects.hpp

output
autoware_perception_msgs/msg/dynamic_object_array.hpp
→
autoware_auto_perception_msgs/msg/predicted_objects.hpp

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: perception/object_recognition/prediction/map_based_prediction/src/map_based_prediction_ros.cpp [hunk 1]
```diff
@@ -250,16 +261,20 @@ void MapBasedPredictionROS::objectsCallback(
...
       tf2::doTransform(pose_orig, pose_in_map, world2map_transform);
-      tmp_object.object.state.pose_covariance.pose = pose_in_map.pose;
+      tmp_object.object.kinematics.pose_with_covariance.pose = pose_in_map.pose;
     }
 
     if (
-      object.semantic.type != autoware_perception_msgs::msg::Semantic::CAR &&
-      object.semantic.type != autoware_perception_msgs::msg::Semantic::BUS &&
-      object.semantic.type != autoware_perception_msgs::msg::Semantic::TRUCK) {
-      tmp_objects_without_map.objects.push_back(tmp_object.object);
```
Comment (yukkysaito) [inline perception/object_recognition/prediction/map_based_prediction/src/map_based_prediction_ros.cpp:262]:
```
using autoware_auto_perception_msgs::msg::ObjectClassification Label;
const auto & label = object.classification.front().label;

if(label == Label::CAR)
```
とかをしてもらったほうがわかりやすいかもです

File: perception/object_recognition/prediction/map_based_prediction/src/map_based_prediction_ros.cpp [hunk 2]
```diff
@@ -250,16 +261,20 @@ void MapBasedPredictionROS::objectsCallback(
...
+      object.classification.front().label !=
+        autoware_auto_perception_msgs::msg::ObjectClassification::CAR &&
+      object.classification.front().label !=
+        autoware_auto_perception_msgs::msg::ObjectClassification::BUS &&
+      object.classification.front().label !=
+        autoware_auto_perception_msgs::msg::ObjectClassification::TRUCK) {
```
Comment (yukkysaito) [inline perception/object_recognition/prediction/map_based_prediction/src/map_based_prediction_ros.cpp:275]: TRAILERがなさそうです

Comment (Sugatyon) [reply perception/object_recognition/prediction/map_based_prediction/src/map_based_prediction_ros.cpp]: @yukkysaito 
対応しました！

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**